erigon-lib/state/merge.go

```diff
@@ -156,40 +134,38 @@ func (dt *DomainRoTx) findMergeRange(maxEndTxNum, maxSpan uint64) DomainRanges {
 func (ht *HistoryRoTx) findMergeRange(maxEndTxNum, maxSpan uint64) HistoryRanges {
 	var r HistoryRanges
 	mr := ht.iit.findMergeRange(maxEndTxNum, maxSpan)
-	r.index, r.indexStartTxNum, r.indexEndTxNum = mr.needMerge, mr.from, mr.to
+	r.index = *mr
+
 	for _, item := range ht.files {
 		if item.endTxNum > maxEndTxNum {
 			continue
 		}
 		endStep := item.endTxNum / ht.h.aggregationStep
 		spanStep := endStep & -endStep // Extract rightmost bit in the binary representation of endStep, this corresponds to size of maximally possible merge ending at endStep
 		span := min(spanStep*ht.h.aggregationStep, maxSpan)
-		start := item.endTxNum - span
-		foundSuperSet := r.indexStartTxNum == item.startTxNum && item.endTxNum >= r.historyEndTxNum
+		startTxNum := item.endTxNum - span
+
+		foundSuperSet := r.history.from == item.startTxNum && item.endTxNum >= r.history.to
```

Comment:
at this point i am not 100% sure since we checking `History` file start/end against found range, but we looked on `index.from` instead. 
changed to history